presto-hive/src/test/java/com/facebook/presto/hive/TestBackgroundSplitLoader.java

```diff
@@ -0,0 +1,276 @@
+                                new DataSize(1.0, GIGABYTE))).getSessionProperties());
+
+        return new BackgroundHiveSplitLoader(
+                TEST_CONNECTOR_ID,
+                table,
+                hivePartitionMetadatas,
+                Optional.empty(),
+                hiveBuckets,
+                connectorSession,
+                new MockHdfsEnvironment(),
+                new NamenodeStats(),
+                new MockDirectoryLister(files),
+                executor,
+                2,
+                0,
+                false);
+    }
+
+    private List<HiveBucket> getBuckets(int bucketCount)
+    {
+        List<HiveBucket> buckets = new ArrayList();
+        for (int i = 0; i < bucketCount; i++) {
+            buckets.add(new HiveBucket(i, bucketCount));
+        }
+        return buckets;
+    }
+
+    private Table getTable()
+    {
+        return getTable(ImmutableList.of());
+    }
+
+    private Table getTable(List<Column> partitionColumns)
+    {
+        Table.Builder tableBuilder = Table.builder();
+        tableBuilder.getStorageBuilder()
+                .setStorageFormat(StorageFormat.create("com.facebook.hive.orc.OrcSerde",
+                        "org.apache.hadoop.hive.ql.io.RCFileInputFormat",
+                        "org.apache.hadoop.hive.ql.io.RCFileInputFormat"))
+                .setLocation("hdfs://VOL1:9000/db_name/table_name")
+                .setSkewed(false)
+                .setBucketProperty(Optional.empty())
+                .setSorted(false);
+
+        return tableBuilder.setDatabaseName("test_dbname")
+                .setOwner("testOwner")
+                .setTableName("test_table")
+                .setTableType(TableType.MANAGED_TABLE.toString())
+                .setDataColumns(ImmutableList.of(
+                        new Column("col1", HIVE_STRING, Optional.empty())))
+                .setParameters(ImmutableMap.of())
+                .setPartitionColumns(partitionColumns)
+                .build();
+    }
+
+    private LocatedFileStatus getLocatedFileStatus(Path path)
+    {
+        return new LocatedFileStatus(0L,
+                false,
+                0,
+                0L,
+                0L,
+                0L,
+                null,
+                null,
+                null,
+                null,
+                path,
+                new BlockLocation[] { new BlockLocation() });
+    }
+
+    private class MockHdfsEnvironment extends HdfsEnvironment
+    {
+        public MockHdfsEnvironment()
+        {
+            super(new HiveHdfsConfiguration(new HdfsConfigurationUpdater(new HiveClientConfig(), new HiveS3Config())),
+                    new HiveClientConfig(),
+                    new NoHdfsAuthentication());
+        }
+
+        @Override
+        public FileSystem getFileSystem(String user, Path path, Configuration configuration)
+        {
+            return new WebHdfsFileSystem();
```

Comment:
Added a new FileSystem implementation, TestingHdfsFileSystem. Fixed in next iteration.